src/main/java/com/startsteps/ecommerceapi/service/CartServiceImpl.java

```diff
@@ -66,14 +66,28 @@ public void addProductToCart(ProductAddRequest request) {
 
            cartProduct.setQuantity(cartProduct.getQuantity() + request.getQuantity());
            cartProduct.setPriceProduct(calculateProductCost(product, cartProduct.getQuantity()));
-           cartProduct.setTotalCost(updateTotalCost(cart));
            cartProductRepository.save(cartProduct); // Ensure you save the updated cartProduct
-
        } else {
            addNewProductToCart(product, cart, request.getQuantity()); // Pass the Product entity, not DTO
        }
-
        reduceProductStock(request.getQuantity(), request.getProductId());
+       getTotalCost(request.getCartId());
+   }
+   @Override
+   public void getTotalCost(Long shoppingCartId){
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByCartId(shoppingCartId)
+                .orElseThrow(()-> new CartIsEmptyException("Cart is not found"));
+       List<CartProduct> cartProducts = cartProductRepository.findCartProductByShoppingCart(shoppingCart);
+       double totalCost = 0.0;
+       for(CartProduct cp : cartProducts){
+           totalCost += cp.getTotalCost();
+       }
+       for(CartProduct cp : cartProducts){
+           cp.setTotalCost(totalCost);
```

Comment:
This logic seems a bit weird to me. You go through a list of all products and create a sum of the total cost, they you set the same total price for every product. Is this what's happening?

Reply:
so each product keeps the total price per product (quantity * product) and the cart's total cost. but I will remove the total cost  per cart in the cart_product entity and keep it just in the shopping cart to avoid misunderstanding :)